libcamera-sys/c_api/camera.cpp

```diff
@@ -67,7 +72,11 @@ const libcamera_control_list_t *libcamera_camera_properties(const libcamera_came
 }
 
 libcamera_camera_configuration_t *libcamera_camera_generate_configuration(libcamera_camera_t *cam, const enum libcamera_stream_role *roles, size_t role_count) {
+#if LIBCAMERA_VERSION_MAJOR == 0 && LIBCAMERA_VERSION_MINOR == 0
     libcamera::StreamRoles roles_vec((libcamera::StreamRole*)roles, (libcamera::StreamRole*)roles + role_count);
+#else
+    std::span roles_vec {(libcamera::StreamRole*)roles, role_count};
```

Comment:
You're probably better off using libcamera::utils::span here - it wouldn't need a #if conditional then.